src/Network/AWS/S3/StreamingUpload.hs

```diff
@@ -144,30 +145,33 @@ streamUpload mChunkSize multiPartUploadDesc =
     multiUpload :: BucketName -> ObjectKey -> Text -> (Int, S) -> m (Maybe CompletedPart)
     multiUpload bucket key upId (partnum, s) = do
       buffer@(PS fptr _ _) <- liftIO $ finaliseS s
-      res <- send $! uploadPart bucket key partnum upId $! toBody $! (HashedBytes $! hash buffer) buffer
+      res <- send env $! newUploadPart bucket key partnum upId $! toBody $! (HashedBytes $! hash buffer) buffer
       let !_ = rwhnf res
       liftIO $ finalizeForeignPtr fptr
-      when (res ^. uprsResponseStatus /= 200) $
+      when (res ^. uploadPartResponse_httpStatus /= 200) $
```

Comment:
This will never match unless you configure the http manager not to throw on non-200.